tools/sqllogictest/sqllogictest.cpp

```diff
@@ -129,6 +129,38 @@ auto ProcessExtraOptions(const std::string &sql, bustub::BustubInstance &instanc
           return false;
         }
         check_options->check_options_set_.emplace(bustub::CheckOption::ENABLE_NLJ_CHECK);
+      } else if (bustub::StringUtil::StartsWith(opt, "ensure:column-pruned")) {
+        auto args = bustub::StringUtil::Split(opt, ":");
+        if (args.size() != 4) {
+          throw bustub::NotImplementedException(fmt::format("unsupported extra option: {}", opt));
+        }
+        auto expected_cols_proj = std::stoi(args[2]);
+        auto expected_cols_agg = std::stoi(args[3]);
+        // find agg & proj plan and test if the output schema has the expected number of columns
+          auto lines = bustub::StringUtil::Split(result.str(), "\n");
+          for (auto &line : lines) {
+            bustub::StringUtil::LTrim(&line);
+            if (bustub::StringUtil::StartsWith(line, "Agg")) {
+              auto cols = bustub::StringUtil::Split(line, "],");
+              if (cols.size() != 3) {
+                fmt::print("Agg plan wrong formatting!\n");
+                return false;
+              }
+              for (int i = 0; i < 2; i++) {
+                if (bustub::StringUtil::Count(cols[i], "\",")+1 != static_cast<size_t>(expected_cols_agg)) {
```

Comment:
```suggestion
                if (bustub::StringUtil::Count(cols[i], "\",")+1 > static_cast<size_t>(expected_cols_agg)) {
```
Maybe this change allows more freedom to let students optimize? Let's say for this optimized plan:
```   
Projection { exprs=["#0.0", "#0.1", "((#0.2+#0.3)+#0.4)"] } | (__subquery#0.__mock_t7.v:INTEGER, __subquery#0.d1:INTEGER, __subquery#0.d2:INTEGER)
    Agg { types=["max", "max", "max", "max"], aggregates=["#0.1", "#0.1", "#0.1", "#0.2"], group_by=["#0.0"] } | (<unnamed>:INTEGER, <unnamed>:INTEGER, <unnamed>:INTEGER, <unnamed>:INTEGER, <unnamed>:INTEGER)
```
Students may optimize further to:
```   
Projection { exprs=["#0.0", "#0.1", "((#0.1+#0.1)+#0.2)"] } | (__subquery#0.__mock_t7.v:INTEGER, __subquery#0.d1:INTEGER, __subquery#0.d2:INTEGER)
    Agg { types=["max", "max"], aggregates=["#0.1", "#0.2"], group_by=["#0.0"] } | (<unnamed>:INTEGER, <unnamed>:INTEGER, <unnamed>:INTEGER)
```
Feel free to correct me!

Reply:
Good catch!